core/src/main/java/org/apache/carbondata/core/datastore/impl/FileFactory.java

```diff
@@ -656,4 +667,55 @@ public static short getDefaultReplication(String path, FileFactory.FileType file
       throws IOException {
     return getCarbonFile(path, fileType).getDefaultReplication(path);
   }
+
+  /**
+   * Write content into specified file path
+   * @param content content to write
+   * @param filePath file path to write
+   * @throws IOException if IO errors
+   */
+  public static void writeFile(String content, String filePath) throws IOException {
+    AtomicFileOperations fileWrite = AtomicFileOperationFactory.getAtomicFileOperations(filePath);
+    BufferedWriter brWriter = null;
+    DataOutputStream dataOutputStream = null;
+    try {
+      dataOutputStream = fileWrite.openForWrite(FileWriteOperation.OVERWRITE);
+      brWriter = new BufferedWriter(new OutputStreamWriter(dataOutputStream,
+          Charset.forName(CarbonCommonConstants.DEFAULT_CHARSET)));
+      brWriter.write(content);
+      brWriter.flush();
+    } catch (IOException ie) {
+      LOGGER.error("Error message: " + ie.getLocalizedMessage());
+      fileWrite.setFailed();
+      throw ie;
+    } finally {
+      CarbonUtil.closeStreams(brWriter);
+      fileWrite.close();
```

Comment:
The fileWrite will not be closed when the line 692 throw IOException.

Reply:
modified like this:
```
      try {
        CarbonUtil.closeStreams(brWriter);
      } finally {
        fileWrite.close();
      }
```